### PR TITLE
Avoid calling wrapper functions with singleton in Kokkos_Cuda_Task.cpp

### PR DESCRIPTION
**Pull request by masterleinad**
Extracted from https://github.com/kokkos/kokkos/pull/6737. This pull request replaces all instances of calling `CudaInternal::singleton` with a wrapper in `Kokkos_CudaTask.hpp` since these places are possibly using the wrong device.

### CODE REVIEW

File: core/src/Cuda/Kokkos_Cuda_Task.hpp [hunk 1]
```diff
@@ -468,10 +465,12 @@ class TaskQueueSpecializationConstrained<
   static void execute(scheduler_type const& scheduler) {
     const int shared_per_warp = 2048;
     const int warps_per_block = 4;
+    const Kokkos::Cuda exec   = Cuda();  // FIXME_CUDA_MULTIPLE_DEVICES
+    const auto& impl_instance = exec.impl_internal_space_instance();
     const int multi_processor_count =
         // FIXME not sure why this didn't work
-        // scheduler.get_execution_space().cuda_device_prop().multiProcessorCount;
-        CudaInternal::singleton().m_deviceProp.multiProcessorCount;
+        // exec.cuda_device_prop().multiProcessorCount;
```
Comment (masterleinad) [inline core/src/Cuda/Kokkos_Cuda_Task.hpp:472]: The `scheduler_type` passed here doesn't have a `get_execution_space()` member function.

File: core/src/Cuda/Kokkos_Cuda_Task.hpp [hunk 2]
```diff
@@ -247,34 +249,30 @@ class TaskQueueSpecialization<SimpleTaskScheduler<Kokkos::Cuda, QueueType>> {
...
+      KOKKOS_IMPL_CUDA_SAFE_CALL(impl_instance->cuda_device_set_limit_wrapper(
+          cudaLimitStackSize, larger_stack_size));
     }
 
     cuda_task_queue_execute<<<grid, block, shared_total, stream>>>(
         scheduler, shared_per_warp);
 
-    KOKKOS_IMPL_CUDA_SAFE_CALL(
-        (CudaInternal::singleton().cuda_get_last_error_wrapper()));
+    KOKKOS_IMPL_CUDA_SAFE_CALL(impl_instance->cuda_get_last_error_wrapper());
```
Comment (dalg24) [inline core/src/Cuda/Kokkos_Cuda_Task.hpp:267]: Will conflict with #6761

Comment (masterleinad) [reply core/src/Cuda/Kokkos_Cuda_Task.hpp]: Rebased.

**Review by dalg24 — APPROVED** [review]

**Comment by masterleinad** [comment]
Retest this please.

**Review by Rombur — APPROVED** [review]

**Comment by dalg24** [comment]
All CUDA builds passed

**Pull request merged**